### PR TITLE
Updating configuration for Github workflow

### PR DESCRIPTION
**Pull request by snake14**
Updating the configuration file that controls our automated builds. Added a cron schedule and updated to PHP 8.2 for the max PHP version for builds.

### CODE REVIEW

**Comment by snake14** [comment]
@AltamashShaikh I reverted all of the library changes and took care of all of the deprecation errors that weren't coming from the libraries. Are you alright with these changes being merged and then addressing the library upgrades later?

File: tests/Framework/MockResponseClient.php [hunk 1]
```diff
@@ -26,9 +26,12 @@ public function __construct(array $config = array())
         $path = PIWIK_INCLUDE_PATH . CapturingGoogleClient::PATH_TO_CAPTURED_DATA_FILE;
         foreach (new \SplFileObject($path) as $line) {
             $decoded = json_decode($line, $isAssoc = true);
+            if (empty($decoded)) {
+                continue;
+            }
 
-            $key = md5(json_encode($decoded[0]));
-            $value = unserialize($decoded[1]);
+            $key = isset($decoded[0]) ? md5(json_encode($decoded[0])) : '';
```
Comment (AltamashShaikh) [inline tests/Framework/MockResponseClient.php]: @snake14 is this failing somewhere ?

Comment (AltamashShaikh) [reply tests/Framework/MockResponseClient.php]: I mean do we need this change ?

**Review by AltamashShaikh — COMMENTED** [review]
@snake14 Changed looks good, I have just 1 concern added comment for that

File: tests/Framework/MockResponseClient.php (hunk 1 above)
Comment (snake14) [reply tests/Framework/MockResponseClient.php]: I can try reverting it. It might be left over from when I was trying to work around the library upgrade.

Comment (AltamashShaikh) [reply tests/Framework/MockResponseClient.php]: yes please try reverting and check

Comment (snake14) [reply tests/Framework/MockResponseClient.php]: Yep. Good call. Looks like it's working fine without those changes.

**Review by AltamashShaikh — APPROVED** [review]

**Pull request merged**